tools/toolchains/mbed_toolchain.py

```diff
@@ -1093,6 +1094,18 @@ def check_and_add_minimal_printf(self, target):
         ):
             self.flags["common"].append("-DMBED_MINIMAL_PRINTF")
 
+    def check_default_lib_supported(self, target, toolchain):
+        """Check and raise an exception if the default_lib value is not supported for this target."""
+        if  hasattr(target, "default_lib"):
+            if (
+                hasattr(target, "supported_c_libs") == False
+                or toolchain not in target.supported_c_libs
+                or target.default_lib not in target.supported_c_libs[toolchain]
```

Comment:
To be case case insensitive in case the user adds support for a target but types "SMALL" instead of "small".
```suggestion
                or target.default_lib.lower() not in target.supported_c_libs[toolchain]
```